README.md

```diff
@@ -1,128 +1 @@
-  2. the `builder` package then generates the grid.Deployment objects.
-
-- ### **Reverting a deployment:**
-
-  1. before applying any change, the deployment manager should first retreive the current state from the nodes `oldState`.
-  2. at first, `currentState` of deployment ids is provided by the user (as `oldDeloymentIDs`).
-  3. every contract deletion or creation, should directly be reflected in the `currentState`.
-  4. if some error happens while applying some change, the deployment manager should revert to its old state using the `currentState` as the `oldDeploymentIDs` and the `oldState` as the `newDeployemnts`.
-
-- ### **Retrieving current state:**
-
-  1. grid3_go users should mainly keep track of contract ids
-  2. the deployment manager should use the provided contract ids to retrieve current state from nodes (using `client` package).
-  
-
-    Example:
-
-     - oldDeploymentIDs :1, 2, 3 - newDeployments: 3, 4, 5
-     - desired state: 3 update, 4 create, 5 create
-     - deploymentManager.deploy(oldDeploymentIDs[1,2,3], newDeployments[3,4,5])
-     - error happens: 1 deleted, 2 not affected, 3 updated, 4 created, 5 not created
-     - currentState: 2, 3, 4
-     - deploymentManager.deploy(currentState[2,3,4], oldDeployments[1,2,3])
-## **grid3_go has the following components:**
-
-- ### **GridClient:**
-
-  - Handles interaction with the chain (Create, Update, Delete).
-  - Handles interaction with the nodes. (Create, Update, Delete).
-  - Be stateless.
-
-    Example:
-
-```go
-type GridClient interface{
-    Create(deployments []grid.Deployment) (uint64, error)
-    Update(deployments []grid.Deployment) error
-    Delete(deploymentIDs []uint64) error
-    Wait() error
-}
-```
-
-- ### **DeploymentManager:**
-
-  - Calculates needed changes between different provided states.
-  - Uses the GridClient to apply needed changes.
-  - Retreives current state of specified deployments.
-
-```go
-type DeploymentManager interface{
-    CalculateChanges(current NodeDeploymentIDMap, new NodeDeploymentMap) ([]create, []update, []delete, error)
-    ApplyChanges(current NodeDeploymentIDMap, new NodeDeplyomentMap) (current NodeDeploymentIDMap,error)
-    GetCurrentDeployments(current NodeDeploymentIDMap) (NodeDeploymentMap, error)
-}
-```
-
-- ### **Loader:**
-
-  - Retreives workload current state from the grid.
-  - Convert from grid types to grid3_go types.
-
-- ### **NodeClient:**
-
-  - Uses gridproxy to get information about nodes, farms, and/or twins.
-  - Uses rmb client (from gridproxy) to interact with nodes.
-
-- ### **Subi:**
-
-  - Exposes an interface to interact with the chain.
-  - Allows mocking substrate-client for testing.
-
-- ### **Builder:**
-
-  - builder package is mainly responsible of:
-
-  1. Expose the needed user friendly builder structs.
-  2. Generate needed grid deployments from builders to be used by the deployer.
+# grid3_client_go
```

Comment:
we forgot to add a simple README file to the repo?
well maybe can be addressed in another PR